xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -5491,8 +5495,25 @@ glusterd_compare_friend_data(dict_t *peer_data, int32_t *status, char *hostname)
         goto out;
     }
 
+    arg = GF_CALLOC(1, sizeof(*arg), gf_common_mt_char);
+    if (!arg) {
+        ret = -1;
+        gf_msg("glusterd", GF_LOG_ERROR, ENOMEM, GD_MSG_NO_MEMORY,
+               "Out Of Memory");
+        goto out;
+    }
+    arg->status_arr = GF_CALLOC(count, sizeof(uint32_t), gf_common_mt_int);
+    if (!arg->status_arr) {
+        ret = -1;
+        gf_msg("glusterd", GF_LOG_ERROR, ENOMEM, GD_MSG_NO_MEMORY,
+               "Out Of Memory");
+        goto out;
+    }
```

Comment:
There's no need to do 2 allocations. We can simply allocate it once:

```c
arg = GF_CALLOC(1, sizeof(*arg) + sizeof(uint64_t) * (count / 64), gf_common_mt_char);
```

The `status_arr` can be declared as the last field of the structre as:

```
uint64_t status_arr[1];
```